cpp/path_module/algorithm/path.cpp

```diff
@@ -1,301 +1,343 @@
-          if (relationship.To().Id() == id) {
-            return true;
-          }
-        }
-        return false;
-      };
+  value = config.At("filterStartNode");
+  config_.filter_start_node = value.IsNull() ? true : value.ValueBool();
 
-      if ((endpoint_is_from && !contains(rel.From().OutRelationships(), rel.To().Id())) ||
-          (!endpoint_is_from && !contains(rel.To().OutRelationships(), rel.From().Id()))) {
-        break;
-      }
+  value = config.At("beginSequenceAtStart");
+  config_.begin_sequence_at_start = value.IsNull() ? true : value.ValueBool();
 
-      path.Expand(rel);
-    }
+  value = config.At("bfs");
+  config_.bfs = value.IsNull() ? false : value.ValueBool();
+}
 
-    auto record = record_factory.NewRecord();
-    record.Insert(std::string(kResultCreate).c_str(), path);
+Path::RelDirection Path::PathHelper::GetDirection(std::string &rel_type) {
+  auto it = config_.relationship_sets.find(rel_type);
+  if (it == config_.relationship_sets.end()) {
+    return RelDirection::kNone;
+  }
+  return it->second;
+}
 
-  } catch (const std::exception &e) {
-    record_factory.SetErrorMessage(e.what());
-    return;
+Path::LabelBools Path::PathHelper::GetLabelBools(const mgp::Node &node) {
+  LabelBools label_bools;
+  for (auto label : node.Labels()) {
+    FilterLabel(label, label_bools);
   }
+  return label_bools;
 }
 
-bool Path::PathSizeOk(const int64_t path_size, const int64_t &max_hops, const int64_t &min_hops) {
-  return (path_size + 1 <= max_hops) && (path_size + 1 >= min_hops);
+bool Path::PathHelper::AreLabelsValid(const LabelBools &label_bools) const {
+  return !label_bools.blacklisted &&
+         ((label_bools.end_node && config_.label_bools_status.end_node_activated) || label_bools.terminated ||
+          (!config_.label_bools_status.termination_activated && !config_.label_bools_status.end_node_activated &&
+           Whitelisted(label_bools.whitelisted)));
 }
 
-bool Path::Whitelisted(const bool &whitelisted, const bool &whitelist_empty) {
-  return (whitelisted || whitelist_empty);
+bool Path::PathHelper::ContinueExpanding(const LabelBools &label_bools, size_t path_size) const {
+  return (static_cast<int64_t>(path_size) <= config_.max_hops && !label_bools.blacklisted && !label_bools.terminated &&
+          (label_bools.end_node || Whitelisted(label_bools.whitelisted)));
 }
-bool Path::ShouldExpand(const LabelBools &labelBools, const LabelBoolsStatus &labelStatus) {
-  return !labelBools.blacklisted && ((labelBools.end_node && labelStatus.end_node_activated) || labelBools.terminated ||
-                                     (!labelStatus.termination_activated && !labelStatus.end_node_activated &&
-                                      Whitelisted(labelBools.whitelisted, labelStatus.whitelist_empty)));
+
+bool Path::PathHelper::PathSizeOk(const int64_t path_size) const {
+  return (path_size <= config_.max_hops) && (path_size >= config_.min_hops);
 }
```

Comment:
Why this part checks max and min hops and in ContinueExpandin we just check max hops?

Reply:
For a path to be appended in results it has to be of size  minhops <= size <= maxhops but if it doesn't satisfy minhops yet we want to still continue expanding

Reply:
Makes sense